src/reaction.rs

```diff
@@ -275,16 +276,28 @@ pub(crate) async fn set_msg_reaction(
     contact_id: ContactId,
     timestamp: i64,
     reaction: Reaction,
+    is_incoming_fresh: bool,
 ) -> Result<()> {
     if let Some((msg_id, _)) = rfc724_mid_exists(context, in_reply_to).await? {
-        set_msg_id_reaction(context, msg_id, chat_id, contact_id, timestamp, reaction).await
+        set_msg_id_reaction(context, msg_id, chat_id, contact_id, timestamp, &reaction).await?;
+
+        if is_incoming_fresh && contact_id != ContactId::SELF && !reaction.is_empty() {
```

Comment:
`contact_id != ContactId::SELF` -- i'd prefer to remove this, AFAIU this code is only for received reactions, not for sent ones, so this can be useful for shared (community) accounts. I.e. absense of this condition shouldn't trigger a notification when the user themselves puts a reaction

Reply:
> I.e. absense of this condition shouldn't trigger a notification when the user themselves puts a reaction

i wanted to protect against notifications sent from another device in a multi-device setup.

are you sure, this is already covered by the other conditions?

Reply:
I was wrong, removing this condition doesn't help for shared accounts because `is_incoming_fresh` means that the reaction is sent from another address. Still, this check is just excessive, `contact_id` cant be `SELF` for _incoming_ messages/reactions

Reply:
What i wanted is that if another device reacts to our (i.e. outgoing) message, we get a notification. This shouldn't affect usual users as people rarely react to their own messages, but would help shared accounts users. So, `is_incoming_fresh` should also be replaced with smth like `is_incoming_fresh || is_outgoing`

Reply:
> This shouldn't affect most users as people rarely react to their own messages

oh, i see this _very_ often, i know quote some ppl doing that on a regular base, eg. adding a 😅 to underline the meaning (as one adds `:)` to the text :)
self-sent-reactions should not trigger a notification - same as self-sent-messages.

let's not optimise for some future of shared accounts in this PR, there are lots of other things wrt notification that need to be adapted when the time comes - and then we probably also have a flag or so.

Reply:
> Still, this check is just excessive, contact_id cant be SELF for incoming messages/reactions

k, if that is true, then i'll remove the check

Reply:
> so, self-sent-reactions should not trigger a notification - same as self-sent-messages.

Forgot about this point, makes sense